### ISSUE TITLE
nasa-veda-prod role has limited bucket access

### ISSUE THREAD

**Issue by abarciauskas-bgse** [2i2c-org/infrastructure#2689]
### Context

https://github.com/2i2c-org/infrastructure/blob/master/terraform/aws/projects/nasa-veda.tfvars defines `hub_cloud_permissions` which limits the buckets that the nasa-veda-prod role (defined in [prod.values.yaml](https://github.com/2i2c-org/infrastructure/blob/master/config/clusters/nasa-veda/prod.values.yaml)) can access.

Is the reason for this a concern about users making request to requester pays buckets? Otherwise, this seems unnecessarily restrictive. By default, I cannot access data in any public bucket on the VEDA 2i2c JHub.  (For example s3://nex-gddp-cmip6)

### Proposal

remove the bucket whitelist, or determine a way to provide that role with access to any bucket that is not requester-pays

**Comment by abarciauskas-bgse** [comment]
@freitagb  @ranchodeluxe @j08lue curious on your thoughts here

**Comment by yuvipanda** [comment]
The primary reason is to make sure we don't accidentally provide access to buckets with sensitive data (particularly tf state) that might happen to be in the same project and don't have additional controls applied. The tfstate for the hub is not in S3 in the account, but since we aren't the exclusive users of this account we aren't sure. 

Happy to turn that into a wildcard! See https://github.com/2i2c-org/infrastructure/pull/2618 for the previous round of this specific conversation with @ranchodeluxe

**Comment by j08lue** [comment]
> buckets with sensitive data (particularly tf state) that might happen to be in the same project

if we are talking buckets in the same AWS account, there is no concern from me regarding wild-carding. We do not have or plan to have buckets with sensitive content in the same project, @abarciauskas-bgse. 

But @freitagb is better positioned to approve this.

**Comment by abarciauskas-bgse** [comment]
This is a good point: "accidentally provide access to buckets with sensitive data (particularly tf state) that might happen to be in the same project". It seem like ensuring no one and no system creates a bucket with sensitive information in that account could be hard to ensure. But I also want to make sure we are enabling users to have direct s3 access to any bucket which permits public access. 🤔

**Comment by yuvipanda** [comment]
> It seem like ensuring no one and no system creates a bucket with sensitive information in that account could be hard to ensure.

Yep, that's my worry! Allowing arbitrary code execution inside an AWS account isn't a very common pattern, so we run into issues like this that aren't super common elsewhere.

I'll wait for @freitagb to chime in!

**Comment by yuvipanda** [comment]
I think being able to access any public s3 bucket is pretty important for doing a lot of research, and this restriction might unnecessarily hobble the hub. So I propose that:

1. An audit is done of the existing S3 buckets in the account, to see if any of them are sensitive. I see some cdk metadata buckets there that would be readable if we allow full s3 read, so would someone in your team be able to do this audit?
2. Once done, we somehow at least write up a policy somewhere that internal S3 buckets here should have more specific permissions, to prevent accidental exposure
3. Once done, we can open up s3 read access.

How does that sound, @freitagb and @abarciauskas-bgse?

**Comment by abarciauskas-bgse** [comment]
This sounds like a good plan to me. Do you have any suggestions on doing the audit or automating enforcement of a bucket policy?

Ideally, we allow the `nasa-veda-prod` role access to any publicly accessible bucket and any "official" VEDA-sanctioned bucket, which at the moment would likely be the current list of buckets in the policy. It doesn't need to have access to any other buckets in that account.

Restricting access to those other buckets seems like a simpler solution than trying to ensure there's no sensitive data in those buckets. I think ideally we are also not storing sensitive information in S3 buckets.

A few ideas about enforcing bucket access for the `nasa-veda-prod` role:
1. I'm pretty sure, but haven't tested it yet, that we could add an ALLOW for wildcard access, an ALLOW for specific buckets in the account AND a DENY for everything else in the account. This seems like the simplest thing to do if it's possible.
2. Use AWS config (or perhaps there is some other AWS policy enforcement tool I'm not aware of) to ensure all buckets except the VEDA data buckets have a DENY policy for this `nasa-veda-prod` role. Something like what is detailed here: https://aws.amazon.com/blogs/security/how-to-use-aws-config-to-monitor-for-and-respond-to-amazon-s3-buckets-allowing-public-access/. this seems cumbersome but definitely feasible.

 I am googling around for s3 scanner for scanning for sensitive information in S3 buckets but haven't found anything promising yet.

**Comment by batpad** [comment]
@abarciauskas-bgse is this something we still want to do? Is there anything I can do to try and help move this along?
